Makefile.Common

```diff
@@ -266,3 +266,43 @@ testifylint-fix:
 gci: $(TOOLS_BIN_DIR)/gci
 	@echo "running $(GCI)"
 	@$(GCI) write -s standard -s default -s "prefix(github.com/open-telemetry/opentelemetry-collector-contrib)" $(ALL_SRC_AND_DOC)
+
+CHANGED_GOLANG_SOURCES?=$(shell git diff main --name-only | grep -E '.*\.go$$' | grep -v -E '.*_test\.go$$')
+.PHONY: for-affected-components
+for-affected-components:
+	@echo "Checking for affected components..."
+	@if [ -z '$(CHANGED_GOLANG_SOURCES)' ]; then \
+		echo "No go source changes detected in shippable code."; \
+	else \
+		cd $(SRC_ROOT); \
+		DEPENDENT_PKGS=$$(echo $(CHANGED_GOLANG_SOURCES) | xargs sed -n 's|^package .* // import "\(.*\)"$$|\1|p' | uniq); \
```

Comment:
This stops at the components directly dependent on the components being changed. In the future we can consider running through all levels of dependent code, eventually hitting the collector itself. However, I wanted to start small and see what I may have missed in this change.